pandas/tseries/common.py

```diff
@@ -45,32 +45,47 @@ def maybe_to_datetimelike(data, copy=False):
         raise TypeError("cannot convert an object of type {0} to a datetimelike index".format(type(data)))
 
     index = data.index
+    name = data.name
+    if is_categorical_dtype(data):
```

Comment:
Not sure if this is better, as it is two instead of one comparison:

```
 if is_categorical_dtype(data):
        orig = data
        data = orig.values.categories
    else:
        orig = None
```
vs
```
orig = data if  is_categorical_dtype(data) else None
if orig is not None:
        data = orig.values.categories
```